app/controllers/bio_eddie_assets_controller.rb

```diff
@@ -3,10 +3,10 @@
 class BioEddieAssetsController < ApplicationController
   include ActiveStorage::SetCurrent
 
-  before_action :load_vars, except: :create
+  before_action :load_vars, except: %i(create bmt_request)
   before_action :load_create_vars, only: :create
 
-  before_action :check_read_permission
+  before_action :check_read_permission, except: %i(update create start_editing bmt_request)
```

Comment:
Prefer `only:`, especially if there are many exceptions (and `only:` is more explicit).

Reply:
read permission checks are default, you should explicitly disable it for specific actions

Reply:
agreed.